scripts/make_report_tables.py

```diff
@@ -0,0 +1,777 @@
+    directions: list[str],
+    distributions: list[str],
+    atk_name: str,
+    atk_metric_name: str,
+) -> list[list[pearsonr]]:
+
+    # Output Array: similarity metrics on rows, attack metrics on columns
+    out = []
+
+    # Counter
+    i = 0
+
+    # Loop over directions (A2B, B2A)
+    for direction in directions:
+        # Filter down runs
+        runs = [run for run in wandb_runs if direction in run.summary.keys()]
+
+        # Loop over distributions (A, B)
+        for distribution in distributions:
+
+            # New column in output
+            out.append([])
+
+            # Get attack metric for ddirection + dist
+            atk_metric = []
+            for run in runs:
+                atk_metric.append(
+                    run.summary[direction][distribution][atk_name][atk_metric_name]
+                )
+
+            # Compute correlations
+            for sim_metric_name in sim_metric_names:
+                sim_metric = [run.summary[sim_metric_name] for run in runs]
+                out[i].append(pearsonr(sim_metric, atk_metric))
+
+            # Iterate
+            i += 1
+
+    # Output
+    return out
+
+
+# H3 Function that takes a correlation table, converts it to a latex table
+# Need to:
+# - Add upper and lower headers - opinionated
+# - Add row names
+# - Round numbers to 2 decimal places
+# - Bold significant numbers
+# - Vary label + title according to attack and metric
+def cor_to_tex(
+    cor: list[list[pearsonr]],
+    atk_name: str,
+    atk_metric_name: str,
+    atk_label: str,
+    atk_metric_label: str,
+    sim_metric_labels: list[str],
+    confidence_level: float = 0.95,
+) -> None:
+    # Get output dimensions
+    ncol = len(cor)
+    nrow = len(sim_metric_labels)
+
+    # Threshold for significance
+    threshold = 1 - confidence_level
+
+    # Table name for latex label + file name
+    tname = f"{atk_name}_{atk_metric_name}"
+
+    # Write into latex table components
+    t_head = (
+        "\\begin{table}[H]\n"
+        "\\centering\n"
+        f"\\caption{{{atk_label} - {atk_metric_label}}}\n"
+        f"\\label{{tab:{tname}}}\n"
+        "\\begin{tabular}{c|c|c|c|c}\n"
+        "Similarity Metric & \\multicolumn{2}{c|}{A to B} & "
+        "\\multicolumn{2}{c}{B to A}\\\\\n\\hline\n"
+        "& A Distribution & B Distribution & A Distribution & "
+        "B Distribution \\\\\n\\hline\n"
+    )
+    t_content = ""
+    extra = 0
+    for i in range(nrow):
+        if sim_metric_labels[i] != "hline":
+            t_content = t_content + sim_metric_labels[i].replace("_", "\\_") + " & "
+            for j in range(ncol):
+                t_content = t_content + write_cor_num(cor[j][i - extra], threshold)
+                if j != (ncol - 1):
+                    t_content = t_content + " & "
+                else:
+                    t_content = t_content + " \\\\\n"
+        if sim_metric_labels[i] == "hline":
+            t_content = t_content + " \\hline\\n "
```

Comment:
We're getting some Latex errors from the H3 table due to the compiler not being able to understand `\hline\n`. I've been trying to find where it is coming from and _think_ it might be this line? Should this read `t_content = t_content + " \\hline\n "` instead (with only one backslash before the n?)

Reply:
Yes, have corrected